src/main/java/com/almightyalpaca/discord/jdabutler/util/MiscUtils.java

```diff
@@ -69,4 +75,21 @@ public static String hastebin(final String text)
             return null;
         }
     }
+
+    public static void announce(TextChannel channel, Role role, Message message, boolean slowmode)
+    {
+        CompletionStage<?> base;
+        if (slowmode)
+            base = channel.getManager().setSlowmode(30).submit();
+        else
+            base = CompletableFuture.completedFuture(null);
+
+        CompletionStage<?> announcement = base
+                .thenRun(() -> role.getManager().setMentionable(true).submit()
+                        .thenRun(() -> channel.sendMessage(message).submit()
+                                .thenRun(() -> role.getManager().setMentionable(false).submit())));
+
+        if (slowmode)
+            announcement.thenRun(() -> channel.getManager().setSlowmode(0).submitAfter(2, TimeUnit.MINUTES));
```

Comment:
This might immediatelly execute after base completes and setMentionable is submitted. Also you could use queueAfter here